cluster-autoscaler/simulator/clustersnapshot/basic.go

```diff
@@ -258,36 +254,36 @@ func (snapshot *BasicClusterSnapshot) IsPVCUsedByPods(key string) bool {
 }
 
 // Fork creates a fork of snapshot state. All modifications can later be reverted to moment of forking via Revert()
-// Forking already forked snapshot is not allowed and will result with an error.
-func (snapshot *BasicClusterSnapshot) Fork() error {
-	if snapshot.forkedData != nil {
-		return fmt.Errorf("snapshot already forked")
-	}
-	snapshot.forkedData = snapshot.baseData.clone()
-	return nil
+func (snapshot *BasicClusterSnapshot) Fork() {
+	forkData := snapshot.getInternalData().clone()
+	snapshot.data = append(snapshot.data, forkData)
 }
 
 // Revert reverts snapshot state to moment of forking.
 func (snapshot *BasicClusterSnapshot) Revert() error {
-	snapshot.forkedData = nil
+	if len(snapshot.data) == 1 {
+		return nil
+	}
+	snapshot.data = snapshot.data[:len(snapshot.data)-1]
 	return nil
 }
 
 // Commit commits changes done after forking.
 func (snapshot *BasicClusterSnapshot) Commit() error {
-	if snapshot.forkedData == nil {
+	if len(snapshot.data) <= 1 {
 		// do nothing
 		return nil
 	}
-	snapshot.baseData = snapshot.forkedData
-	snapshot.forkedData = nil
+	lastData := snapshot.data[len(snapshot.data)-1]
+	snapshot.Revert()
```

Comment:
nit: I think direct pop from the list would be much more readable, I found this very confusing when skimming through the code (I'm not saying this is incorrect, just surprising to me).